src/vmm_core.ml

```diff
@@ -375,28 +395,63 @@ module Unikernel = struct
     let hex_digest = Ohex.encode unikernel.digest in
     Fmt.pf ppf "pid %d@ taps %a (block %a) cmdline %a digest %s"
       unikernel.pid
-      Fmt.(list ~sep:(any ", ") string) unikernel.taps
+      Fmt.(list ~sep:(any ", ") (pair ~sep:(any ": ") string Macaddr.pp)) unikernel.taps
       Fmt.(list ~sep:(any ", ") pp_block) unikernel.config.block_devices
       Fmt.(array ~sep:(any " ") string) unikernel.cmd
       hex_digest
 
+  type block_info = {
+    unikernel_device : string ;
+    host_device : string ;
+    sector_size : int ;
+    size : int ;
+  }
+
+  let pp_block_info ppf { unikernel_device ; host_device ; sector_size ; size } =
+    Fmt.pf ppf "%s -> %s (sector-size: %u bytes, size %u MB)"
+      unikernel_device host_device sector_size size
+
+  type net_info = {
+    unikernel_device : string ;
+    host_device : string ;
+    mac : Macaddr.t ;
+  }
+
+  let pp_net_info ppf { unikernel_device ; host_device ; mac } =
+    Fmt.pf ppf "%s -> %s (mac %a)" unikernel_device host_device
+      Macaddr.pp mac
+
   type info = {
     typ : typ ;
     fail_behaviour : fail_behaviour;
     cpuid : int ;
     memory : int ;
-    block_devices : (string * string option * int option) list ;
-    bridges : (string * string option * Macaddr.t option) list ;
+    block_devices : block_info list ;
+    bridges : net_info list ;
     argv : string list option ;
     digest : string ;
     started : Ptime.t ;
   }
 
-  let info t =
+  let info block_size t =
     let cfg = t.config in
+    let block_devices =
+      List.map (fun (unikernel_device, host_device, sector_size) ->
+          let host_device = Option.value ~default:unikernel_device host_device
+          and sector_size = Option.value ~default:512 sector_size (* TODO: this is the default in solo5-hvt at the moment *)
```

Comment:
Possibly a total hack, but we can perhaps match on  the ABI version and use that to determine it's 512. Then we have a migration path if we should want to change the default in solo5 at some point (--> we require an abi change). The downside is I don't think it would be an /actual/ ABI change to change the default...